lib/FunctionUtils.lua

```diff
@@ -34,6 +34,50 @@ function FunctionUtils.memoize(fn, serializeArgs)
 	return proxyFunction
 end
 
+function FunctionUtils.timeout(fn, secondsDelay)
+	local cleared = false
+	delay(
+		secondsDelay,
+		function()
+			if not cleared then
+				fn()
+			end
+		end
+	)
+	return {
+		clear = function()
+			cleared = true
+		end
+	}
+end
+
+function FunctionUtils.interval(fn, secondsDelay)
+	local timeout
+	local callTimeout
+	callTimeout = function()
+		timeout =
+			FunctionUtils.timeout(
+			function()
+				fn()
+				callTimeout()
+			end,
+			secondsDelay
+		)
+	end
+	callTimeout()
+
+	return {
+		clear = function()
+			timeout:clear()
+		end
+	}
+end
+
+function FunctionUtils.intervalNow(fn, secondsDelay)
```

Comment:
Is this function worth having? From the name alone, I wouldn't have known what it did. Add in the time to look up this function, and I might as well have just read the two lines below.